### PR TITLE
Save numbers to vars as is

### PR DESCRIPTION
**Pull request by Destroy666x**
The PR:
- makes it so that e.g. incrementing an int in a variable keeps it as int, since previous `to_string` used `sprintf` under the hood and added a random number of zeros after `.`, depending on compiler. This should be much more consistent. Even better would be `std::format` from what I read, but it's only C++20 and that was impossible for now.
- checks for all limits in the `GetDouble`/`GetInt` functions to avoid some false conversions. Only upper limit of double was checked for some reason earlier. This is still far from perfect as you can't go up to max, but shouldn't matter too much. Also, bigger numbers support would be nice with different types, but I'll create another issue for that.

### CODE REVIEW

File: lib/macro/macro-action-variable.cpp [hunk 1]
```diff
@@ -68,6 +68,7 @@ static void modifyNumValue(Variable &var, double val, const bool increment)
 	if (!current.has_value()) {
 		return;
 	}
+
```
Comment (WarmUpTill) [inline lib/macro/macro-action-variable.cpp:71]: Sorry for being pedantic, but is that change really needed? :)

Comment (Destroy666x) [reply lib/macro/macro-action-variable.cpp:71]: This one in particular or adding some spacing for readability in general?

Comment (Destroy666x) [reply lib/macro/macro-action-variable.cpp:71]: It's in the increment function, which is directly related. There were even changes before, but this was the only one I left in the end.

Comment (WarmUpTill) [reply lib/macro/macro-action-variable.cpp:71]: >It's in the increment function, which is directly related.

This is not what I meant.
Adding an empty line to `macro-action-variable.cpp` is not related to the change `Check for lower/upper GetDouble/GetInt limits`.

>There were even changes before, but this was the only one I left in the end. Do you want for me to get rid of this single line addition?

Feel free to put those types of changes in a separate commit or even a separate PR, if you like. :)
To me this change simply does not fit into `Check for lower/upper GetDouble/GetInt limits` (even with `+ small refactors` added at the end).

I am probably just being overly cautious, but I just want to be sure, that in case any issues pop up, it is as easy as possible to track down what was changed and why.
Having those style changes in `macro-action-variable.cpp` mixed in with `Check for lower/upper GetDouble/GetInt limits` might hinder that.

I hope you understand! :)

And I am not trying to be dismissive or unappreciative of you putting in time to improve the plugin!
I value that a lot - so thank you!

Comment (Destroy666x) [reply lib/macro/macro-action-variable.cpp:71]: I understand ofc, but whitespace change does nothing in terms of debugging IMO. I'll move it to another commit.
